app/models/planning_application.rb

```diff
@@ -1075,6 +1076,12 @@ def changed_to_prior_approval?
     application_type_changed? && application_type.name == "prior_approval"
   end
 
+  def ensure_reporting_is_correct
+    return if regulation_3
+
+    self.regulation_4 = false if !regulation_3
```

Comment:
It might not be necessary to have both these lines, I think? Don't need to check whether `regulation_3` is false because we've already returned if it's true.